src/m365/spo/commands/homesite/homesite-set.ts

```diff
@@ -1,102 +1,148 @@
       }
 
       const spoAdminUrl = await spo.getSpoAdminUrl(logger, this.debug);
+
+      const configuration: any = {};
+      if (args.options.vivaConnectionsDefaultStart !== undefined) {
+        configuration.IsVivaConnectionsDefaultStartPresent = true;
+        configuration.vivaConnectionsDefaultStart = args.options.vivaConnectionsDefaultStart;
+      }
+      if (args.options.draftMode !== undefined) {
+        configuration.IsInDraftModePresent = true;
+        configuration.isInDraftMode = args.options.draftMode;
+      }
+      if (args.options.audienceIds !== undefined) {
+        configuration.IsAudiencesPresent = true;
+        configuration.Audiences = args.options.audienceIds.split(',').map(id => id.trim());
+      }
+      if (args.options.audienceNames !== undefined) {
+        configuration.IsAudiencesPresent = true;
+        configuration.Audiences = args.options.audienceNames.trim() === '' ? [] : await this.transformAudienceNamesToIds(args.options.audienceNames);
+      }
+      if (args.options.targetedLicenseType !== undefined) {
+        configuration.IsTargetedLicenseTypePresent = true;
+        configuration.TargetedLicenseType = this.convertTargetedLicenseTypeToNumber(args.options.targetedLicenseType);
+      }
+      if (args.options.order !== undefined) {
+        configuration.IsOrderPresent = true;
+        configuration.Order = args.options.order;
+      }
+
       const requestOptions: CliRequestOptions = {
-        url: `${spoAdminUrl}/_api/SPO.Tenant`,
+        url: `${spoAdminUrl}/_api/SPO.Tenant/UpdateTargetedSite`,
         headers: {
           accept: 'application/json;odata=nometadata',
           'content-Type': 'application/json'
         },
         responseType: 'json',
         data: {
-          sphSiteUrl: args.options.siteUrl
+          siteUrl: args.options.siteUrl,
+          configurationParam: configuration
         }
       };
 
-      if (args.options.vivaConnectionsDefaultStart !== undefined) {
-        requestOptions.url += '/SetSPHSiteWithConfiguration';
-        requestOptions.data.configuration = { vivaConnectionsDefaultStart: args.options.vivaConnectionsDefaultStart };
-      }
-      else {
-        requestOptions.url += '/SetSPHSite';
-      }
-
-      const res = await request.post<{ value: string; }>(requestOptions);
-      await logger.log(res.value);
+      const res = await request.post(requestOptions);
+      await logger.log(res);
     }
     catch (err: any) {
       this.handleRejectedODataJsonPromise(err);
     }
   }
-}
 
+  private convertTargetedLicenseTypeToNumber(licenseType: string): number {
+    const licenseTypeMap: Record<string, number> = {
+      'everyone': 0,
+      'frontLineWorkers': 1,
+      'informationWorkers': 2
+    };
+    return licenseTypeMap[licenseType];
+  }
+
+  private async transformAudienceNamesToIds(audienceNames: string): Promise<string[]> {
+    const names = audienceNames.split(',').map(name => name.trim());
```

Comment:
Just a small enhancement, right now we're looping twice over the array. First to trim all spaces, later to get the ID of the group.
We could do this in one loop, by just removing the trim here and adding it when we call the `getGroupIdByDisplayName` function.